src/client/java/daos-java/src/main/native/io_daos_dfs_DaosFsClient.c

```diff
@@ -868,6 +991,51 @@ Java_io_daos_dfs_DaosFsClient_dfsWrite(JNIEnv *env, jobject client,
 	return len;
 }
 
+static int
+update_ret_code(void *udata, daos_event_t *ev, int ret)
+{
+	dfs_desc_t *desc = (dfs_desc_t *)udata;
+	char *desc_buffer = desc->ret_buf_address;
+
+	memcpy(desc_buffer, &ret, 4);
+	ev->ev_error = 0;
+}
+
+JNIEXPORT void JNICALL
+Java_io_daos_dfs_DaosFsClient_dfsWriteAsync(JNIEnv *env, jobject client,
+	jlong dfsPtr, jlong objId, jlong descBufAddress)
+{
+    dfs_t *dfs = *(dfs_t **)&dfsPtr;
+    dfs_obj_t *file = *(dfs_obj_t **)&objId;
+    char *buf = (char *)descBufAddress;
+    uint64_t offset;
+    uint64_t len;
+    dfs_desc_t *desc;
+    daos_event_t *event;
+    int rc;
+
+    decode_dfs_desc(buf, &desc, &event, &offset, &len);
+    rc = daos_event_register_comp_cb(event,
+    			update_ret_code, desc);
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs